internal/functions/efgs/upload-batch.go

```diff
@@ -65,7 +66,12 @@ func uploadAndRemoveBatch(ctx context.Context, uploadConfig *uploadConfig, now t
 	for _, batchDbKeys := range batches {
 		var diagnosisKeys []*efgsapi.DiagnosisKey
 		for _, k := range batchDbKeys {
-			diagnosisKeys = append(diagnosisKeys, k.ToData())
+			diagnosisKey := k.ToData()
+			if diagnosisKey == nil {
+				logger.Errorf("DiagnosisKey %s is invalid", k.KeyData)
+			} else {
+				diagnosisKeys = append(diagnosisKeys, diagnosisKey)
+			}
```

Comment:
When you `panic` the code instead of returning `nil`, you can revert this change.